### ISSUE TITLE
kickstart script `--local-files` option overhaul.

### ISSUE THREAD

**Issue by Ferroin** [netdata/netdata#10695]
The current `--local-files` option is complicated, error-prone, and lacks support for a bunch of the things we need to download.

This really needs to be rebuilt almost from scratch at this point.

The proposed implementation would be:

* One option each for specifying each thing we need to download. This allows users to cache certain things locally if they want.
* One option to point to a directory with all the files with appropriate names.
* One option to generate the above mentioned directory, which can then be copied to another system for installs.
* Proper handling of any of the files as ‘optional’ (if missing checksum files, just ignore them, if missing other things, attempt to download them).
* All the downloads consolidated in _one_ place, possibly as a separate script, and ideally run in parallel in some way.

**Comment by Ferroin** [comment]
Will be resolved by https://github.com/netdata/product/issues/2315.

**Comment by wahlfeld** [comment]
> Will be resolved by https://github.com/netdata/product/issues/2315.

This link doesn't work for me.

Also, [the docs](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/offline) don't appear to be up to date, as the option '--local-files' doesn't exist anywhere.

**Comment by Ferroin** [comment]
Closing this as it’s no longer relevant. https://github.com/netdata/netdata/pull/12809 will add support for offline installs with static builds in the new kickstart script. We have no plans currently to implement such support for native packages (users should use their distro’s native mechanisms for this) or local builds (the required changes for this are complex and far-reaching, and it’s an uncommon enough install method now that we do not currently consider it worth working on).
